### PR TITLE
oio/cli: add oio-cluster command into openio client

### PR DESCRIPTION
**Pull request by Crashdown59**
New commands : 
openio cluster list --local -> show all local services of cluster 
openio cluster unlock <service> <addr> (--score <score>) -> unlock the service given in the command with the score <score> 
openio cluster local conf -> print to stdout the namespace configuration values locally configured
openio clear flush <service> -> flush all services of the service type

### CODE REVIEW

File: oio/cli/admin/cluster.py [hunk 1]
```diff
@@ -56,8 +64,111 @@ def take_action(self, parsed_args):
...
+            for srv in data:
+                    # to change
+                infos_srv = {'type': srv_type,
+                             'addr': srv['addr'],
+                             'score': score}
+                if addr == srv['addr']:
+                    try:
+                        self.app.client_manager.admin.cluster_unlock_score(
+                            infos_srv)
+                        print "Service " + service_desc + \
```
Comment (jkasarherou) [inline oio/cli/admin/cluster.py]: Please do not use `print`

File: oio/cli/admin/cluster.py [hunk 2]
```diff
@@ -56,8 +64,111 @@ def take_action(self, parsed_args):
...
+
+    def take_action(self, parsed_args):
+
+        self.log.debug('take_action(%s)', parsed_args)
+        data = self.app.client_manager.admin.cluster_info()
+        sds_conf = load_namespace_conf(data['ns']) or {}
+        output = list()
+        for k in sds_conf:
+            v = sds_conf[k]
+            if v.find("127.0.0.1") != -1:
```
Comment (jkasarherou) [inline oio/cli/admin/cluster.py]: how can you expect this to work?

File: oio/cli/admin/cluster.py [hunk 3]
```diff
@@ -56,8 +64,109 @@ def take_action(self, parsed_args):
                 addr = srv['addr']
                 up = tags.get('tag.up', 'n/a')
                 score = srv['score']
-                results.append((srv_type, addr, volume, location,
-                                slots, up, score))
+                if not local or addr.find('127.0.0.1') != -1:
```
Comment (jkasarherou) [inline oio/cli/admin/cluster.py]: still nope

Comment (jfsmig) [inline oio/cli/admin/cluster.py]: @Crashdown59 : The architecture part aware of local service is the registration point, i.e. the proxy. Whatever, knowing this is not the client's job. Listing the local services has a dedicated route. please consider reading https://github.com/open-io/oio-sds/wiki/OpenIO-Proxyd-API-Reference#list-local-services :)

File: oio/cli/admin/cluster.py [hunk 4]
```diff
@@ -61,3 +63,141 @@ def take_action(self, parsed_args):
         columns = ('Type', 'Id', 'Volume', 'Location', 'Slots', 'Up', 'Score')
         result_gen = (r for r in results)
         return columns, result_gen
+
+
+class ClusterLocalList(lister.Lister):
+    """Cluster local list"""
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: This description is a bit light.
> List local services
would be more expressive.

File: oio/cli/admin/cluster.py [hunk 5]
```diff
@@ -61,3 +63,141 @@ def take_action(self, parsed_args):
...
+
+    log = logging.getLogger(__name__ + '.ClusterLocalList')
+
+    def get_parser(self, prog_name):
+        parser = super(ClusterLocalList, self).get_parser(prog_name)
+        parser.add_argument(
+            'srv_types',
+            metavar='<srv_types>',
+            nargs='?',
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: You should pass `nargs='*'`.

File: oio/cli/admin/cluster.py [hunk 6]
```diff
@@ -61,3 +63,141 @@ def take_action(self, parsed_args):
...
+            help='Service Type(s)')
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug('take_action(%s)', parsed_args)
+        results = []
+        srv_types = parsed_args.srv_types
+        data = self.app.client_manager.admin.cluster_local_list()
+        for srv in data:
+            print srv
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: You forgot a `print` statement.

File: oio/cli/admin/cluster.py [hunk 7]
```diff
@@ -61,3 +63,141 @@ def take_action(self, parsed_args):
...
+                results.append((srv_type, addr, volume, location,
+                                slots, up, score))
+            columns = ('Type', 'Id', 'Volume', 'Location',
+                       'Slots', 'Up', 'Score')
+            result_gen = (r for r in results)
+        return columns, result_gen
+
+
+class ClusterUnlockService(command.Command):
+    """unlock score of the service given"""
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: Don't speak about _the service given_, this is the role of the auto-generated usage.
And don't forget the capital letter.

File: oio/cli/admin/cluster.py [hunk 8]
```diff
@@ -61,3 +63,141 @@ def take_action(self, parsed_args):
...
+
+    def get_parser(self, prog_name):
+        parser = super(ClusterUnlockService, self).get_parser(prog_name)
+        parser.add_argument(
+            'srv_type',
+            metavar='<srv_type>',
+            help='Service Type')
+        parser.add_argument(
+            'addr',
+            metavar='<addr_service>',
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: The service of the address? Or the address of the service?

File: oio/cli/admin/cluster.py [hunk 9]
```diff
@@ -61,3 +63,141 @@ def take_action(self, parsed_args):
...
+
+    def take_action(self, parsed_args):
+        default_value = -1
+        self.log.debug('take_action(%s)', parsed_args)
+        try:
+            score = int(parsed_args.score) or default_value
+        except Exception:
+            score = default_value
+        addr = parsed_args.addr
+        data_cluster = self.app.client_manager.admin.cluster_info()
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: The cluster of data or data about the cluster?
Why don't you keep `cluster_info` as the variable name?

File: oio/cli/admin/cluster.py [hunk 10]
```diff
@@ -61,3 +63,141 @@ def take_action(self, parsed_args):
...
+        srv_type = parsed_args.srv_type
+        service_desc = "[%s|%s|%s]" % (data_cluster["ns"],
+                                       srv_type,
+                                       addr)
+        data = self.app.client_manager.admin.cluster_list(
+            srv_type)
+        for srv in data:
+            # to change
+            infos_srv = {'type': srv_type,
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: What did I say about *information*?

File: oio/cli/admin/cluster.py [hunk 11]
```diff
@@ -61,3 +63,141 @@ def take_action(self, parsed_args):
...
+                try:
+                    self.app.client_manager.admin.cluster_unlock_score(
+                        infos_srv)
+                    self.log.warn("Service " + service_desc +
+                                  " has been successfully unlocked")
+                    return
+                except Exception as e:
+                    raise Exception('service unlock error: ' + str(e))
+        self.log.error('Failed to set score of service ' +
+                       service_desc + ' : Invalid Service Description')
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: Don't put spaces before semicolons.

File: oio/cli/admin/cluster.py [hunk 12]
```diff
@@ -61,3 +63,139 @@ def take_action(self, parsed_args):
...
+            'srv_type',
+            metavar='<srv_type>',
+            help='Service Type')
+        parser.add_argument(
+            'srv_addr',
+            metavar='<srv_addr>',
+            help='addr of the service'
+        )
+        parser.add_argument(
+            '--score',
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: This option has no effect.

File: oio/cli/admin/cluster.py [hunk 13]
```diff
@@ -61,3 +63,130 @@ def take_action(self, parsed_args):
...
+            help='Service Type')
+        parser.add_argument(
+            'srv_addr',
+            metavar='<srv_addr>',
+            help='addr of the service'
+        )
+        return parser
+
+    def take_action(self, parsed_args):
+        default_value = -1
```
Comment (fvennetier) [inline oio/cli/admin/cluster.py]: This variable is never used (the tests failed because of this).

**Pull request merged**